src/Servers/HttpSys/src/AsyncAcceptContext.cs

```diff
@@ -2,121 +2,132 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
-using System.Diagnostics.CodeAnalysis;
 using System.Threading;
 using System.Threading.Tasks;
-using Microsoft.AspNetCore.Http;
+using System.Threading.Tasks.Sources;
 using Microsoft.AspNetCore.HttpSys.Internal;
 
 namespace Microsoft.AspNetCore.Server.HttpSys
 {
-    internal unsafe class AsyncAcceptContext : TaskCompletionSource<RequestContext>, IDisposable
+    internal unsafe class AsyncAcceptContext : IValueTaskSource<RequestContext>, IDisposable
     {
-        internal static readonly IOCompletionCallback IOCallback = new IOCompletionCallback(IOWaitCallback);
+        private static readonly IOCompletionCallback IOCallback = IOWaitCallback;
+        private readonly PreAllocatedOverlapped _preallocatedOverlapped;
+
+        // mutable struct; do not make this readonly
+        private ManualResetValueTaskSourceCore<RequestContext> _tcs = new()
```

Comment:
Nit about the naming of `_tcs`.

It matches well for TaskCompletionSource, but for the ManualResetValueTaskSourceCore...I'd name it `_mrvts` or maybe `_mrvtsc` or `_valueTaskSource` or ...